scipy/stats/tests/test_hypotests.py

```diff
@@ -1236,3 +1236,235 @@ def test_same_input(self):
         # check exact p-value
         res = cramervonmises_2samp(x[:4], x[:4])
         assert_equal((res.statistic, res.pvalue), (0.0, 1.0))
+
+
+class TestTukeyHSD:
+
+    data_same_size = ([24.5, 23.5, 26.4, 27.1, 29.9],
+                      [28.4, 34.2, 29.5, 32.2, 30.1],
+                      [26.1, 28.3, 24.3, 26.2, 27.8])
+    data_diff_size = ([24.5, 23.5, 26.28, 26.4, 27.1, 29.9, 30.1, 30.1],
+                      [28.4, 34.2, 29.5, 32.2, 30.1],
+                      [26.1, 28.3, 24.3, 26.2, 27.8])
+    extreme_size = ([24.5, 23.5, 26.4],
+                    [28.4, 34.2, 29.5, 32.2, 30.1, 28.4, 34.2, 29.5, 32.2,
+                     30.1],
+                    [26.1, 28.3, 24.3, 26.2, 27.8])
+
+    sas_same_size = """
```

Comment:
Personally, I like this a lot better because you can really easily see the connection between the output of the reference implementation (SAS/Matlab/Minitab/R) and the test. You also don't have to separately include the output of the reference implementation in a comment. 

I seem to remember there were some extra characters (e.g. "-", ",") in some of these that it looks like you've removed? If the lines will fit within the 80 char limit if you literally copy-paste without removing those characters (and maybe including the header row, too), that would even better. But thanks for changing this.

Reply:
I will add the header rows back in for R and SAS. Matlab was just cells with no labels when I viewed the result variable. 

I can also add the dashes back in and just strip them out. 

For R, I replaced the "L-H", "L-M", and "H-M" labels with integers, but it would require some reworking to get letters to correspond to the numbers, so I think its cleaner to just leave em.